### ISSUE TITLE
Team Workspaces rolled out to BETA 🎉🎉

### ISSUE THREAD

**Issue by sachinjain024** [requestly/requestly#234]
**TLDR--** Collaboration is one of the most requested feature in Requestly since setting up so many rules individually is quite painful and time-consuming. Therefore, we have built a Team Workspace feature using which the rules of all the members will be synced and Individuals can turn on/off rules as per their needs.

-------------------------------------------------------

:point_right: UPDATE: Team Workspace feature has been rolled out to BETA, Signup/Signin with your Business emails and readily create/switch workspaces to collaborate with different teams. 

Report Bugs & Feature Requests at [Github Discussions](https://github.com/requestly/team-workspaces/discussions)

-------------------------------------------------------

## About Team Workspace feature
Requestly workspaces enable you to organize and work together on projects with your team. Within each workspace you can share [Rules](https://docs.requestly.io/getting-started/modify-http-request-using-rules), [Mocks](https://docs.requestly.io/getting-started/features/mock-server), [Session Recordings](https://requestly.io/feature/session-recording/) and other Requestly features. 

Collaboration is helpful in so many ways:

- Devs <--> Devs (e.g. Sharing rules related to their environment switching, debugging API responses, etc)
- Devs <--> QAs (e.g. Devs share rules with QAs to test their new features without deploying on production)
- Devs <--> Product Manager & Designers (e.g. Getting their feedback/design review on the new feature changes)
- Integration Team <--> Customers (e.g. Demoing how their product works on Customer Sites pre-installation & Customers can also play around with the product)
- Solutions Consultants <--> Customers (e.g. Demoing how the new feature works and getting their approval)

In all cases, right now the other party needs to install Requestly and set up the rules themselves. With workspaces, they just have to install Requestly and get added to the workspace.

### To create a new workspace:

1. Select Workspaces in the sidebar. Select Create Workspace to add a new workspace. You can also search for existing workspaces using the search bar or select one from the dropdown list.
![Screenshot 2022-11-17 at 3 47 36 PM](https://user-images.githubusercontent.com/29792913/202420386-f297b4c1-dbc7-4847-8bdb-af12f78b5e9a.png)

2. Enter a Name and for your workspace and tap Create Workspace

3. Requestly will switch to your new workspace when you create it. You can move between workspaces at any time by selecting from the dropdown Workspaces list in the upper left.

4. Now that you have learned how to create a workspace, you can use them to collaborate with your team. To learn how to invite other users to join your new workspace, visit [Using and managing workspaces](https://docs.requestly.io/other/premium-subscription#for-teams).
 
### ❓ Does it change my existing workflow?

When you first open Requestly, you will be in your default personal workspace. You can create more workspaces for your personal use and to work with teammates. To create more workspaces, you need to sign in to your Requestly account.

### ❓Will my private rules also run if I am on a team workspace?

No. Requestly will only run the rules from the selected workspace. 

### ❓Can I invite my customers (external domain emails) to a workspace?

Yes. Right now there are no restrictions on who can use the team workspace. (Personal emails are not allowed at the moment) We are rolling this out gradually to select companies. If you are interested, please apply via https://forms.gle/SPf9MqDcM5NAb1ETA

### ❓Any Nuances I should be aware of?
Rules are synced whenever a user visits the Requestly page (i.e. [app.requestly.io/rules](http://app.requestly.io/rules)). So if you have made any changes to the rules, the other party has to open the app to sync the latest changes. 

### ❓What if someone updates any working rule by mistake?

We have already implemented audit logs, so you will be able to see who updated the rule definitions and you can easily revert to any previous rule definition at any time.

### Future updates

- Access Control - You will be able to add team members who can only use the rules but not update the rule definitions.
- Notifications - When anyone makes changes to rules in team workspaces, we don’t notify them right now. We will provide better notifications in-app, email, and slack alerts.

Happy to discuss this further and hear all your thoughts on this. Very excited about this most-awaited feature of Requestly.

----------

Apply for early access here - [Requestly Team Workspace - Early Access](https://forms.gle/SPf9MqDcM5NAb1ETA)

**Comment by chinchang** [comment]
This sounds really exciting...just what we need in a team! Applied!
